apps/cowswap-frontend/src/legacy/hooks/useWrapCallback.ts

```diff
@@ -113,14 +113,14 @@ async function wrapContractCall(
   const estimatedGas = await wethContract.estimateGas.deposit({ value: amountHex }).catch(_handleGasEstimateError)
   const gasLimit = calculateGasMargin(estimatedGas)
 
-  const network = await wethContract.provider.getNetwork()
-  if (network.chainId !== chainId) {
+  const network = await (wethContract.provider as JsonRpcProvider).send('eth_chainId', [])
+  if (+network !== chainId) {
     throw new Error('Wallet chain differs from order params.')
```

Comment:
Out of scope, but still.
Can you add both chainIds and account address to the error please?
We would definitely need those details if we face the issue again.
Also, have ensured that the error gets into Sentry logs?

Reply:
Will update the message to include the chains.
Have not checked sentry.

Reply:
Why do we need to rely on casting?

Reply:
Because the instance from the contract [is `Provider`](https://docs.ethers.org/v5/api/providers/provider/#BaseProvider)
![image](https://github.com/user-attachments/assets/e1bf2ec8-618f-4818-bb11-b92e1ce659bc)

Which doesn't have the [`send` method in the interface](https://docs.ethers.org/v5/api/providers/jsonrpc-provider/#JsonRpcProvider-send), present in the JsonRpcProvider

Reply:
https://github.com/cowprotocol/cowswap/pull/5354/commits/8553f9bc875949b4819cc55c204811d4907700aa

Reply:
@shoom3301 regarding sentry, apparently it's captured already: https://cowprotocol.sentry.io/issues/6253626499/?project=5905822

See this PR comment https://github.com/cowprotocol/cowswap/pull/5354#issuecomment-2621368355

Reply:
So we assume the provider is always JSON provider? Should we assert it somehow (and throw if its not), or should we default to the provider method getChainId?  (probably best is to throw)

How can we be sure the provider is JsonRpcSigner?